runtime/lua/vim/_editor.lua

```diff
@@ -1010,18 +1010,54 @@ end
 function vim._init_default_mappings()
   -- mappings
 
-  --@private
+  ---@private
+  local function region_to_text(region)
+    local text = ''
+    local maxcol = vim.v.maxcol
+    for line, cols in pairs(region) do
+      local endcol = cols[2] == maxcol and -1 or cols[2]
+      local chunk = vim.api.nvim_buf_get_text(0, line, cols[1], line, endcol, {})[1]
+      text = ('%s%s\n'):format(text, chunk)
+    end
+    return text
+  end
+
+  ---@private
+  local function _search_for_visual_selection(search_prefix)
+    if search_prefix ~= '/' and search_prefix ~= '?' then
+      return
+    end
+    -- Escape these characters
+    local replacements = {
+      [search_prefix] = [[\]] .. search_prefix,
+      [ [[\]] ] = [[\\]],
+      ['\t'] = [[\t]],
+      ['\n'] = [[\n]],
+    }
+    local pattern = '[' .. table.concat(vim.tbl_keys(replacements), '') .. ']'
+    local region = vim.region(0, "'<", "'>", vim.fn.visualmode(), true)
+    local visual_selection = region_to_text(region)
+    local escaped_visual_selection = string.gsub(visual_selection, pattern, replacements)
+    local search_cmd = search_prefix .. [[\V]] .. escaped_visual_selection .. '\n'
+    vim.api.nvim_feedkeys(search_cmd, 'nx', true)
+  end
+
+  ---@private
   local function map(mode, lhs, rhs)
-    vim.api.nvim_set_keymap(mode, lhs, rhs, { noremap = true, desc = 'Nvim builtin' })
+    vim.keymap.set(mode, lhs, rhs, { noremap = true, desc = 'Nvim builtin' })
```

Comment:
```suggestion
    vim.keymap.set(mode, lhs, rhs, { desc = 'Nvim builtin' })
```
`vim.keymap.set` does not support `noremap` (and `remap=false` is the default anyway)